MyOnboarding/src/MyOnboardingApp.Services/Services/UpdateItemService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Threading.Tasks;
+using MyOnboardingApp.Contracts.Generators;
+using MyOnboardingApp.Contracts.Models;
+using MyOnboardingApp.Contracts.Repository;
+using MyOnboardingApp.Contracts.Services;
+using MyOnboardingApp.Contracts.Validation;
+
+namespace MyOnboardingApp.Services.Services
+{
+    internal class UpdateItemService : AbstractUpsertService, IUpdateItemService
+    {
+        private readonly ITodoListRepository _repository;
+        private readonly IDateTimeGenerator _dateTimeGenerator;
+
+
+        public UpdateItemService(ITodoListRepository repository, IDateTimeGenerator dateTimeGenerator, IInvariantValidator<TodoListItem> validator)
+            : base(validator)
+        {
+            _dateTimeGenerator = dateTimeGenerator;
+            _repository = repository;
+        }
+
+
+        public async Task<IItemWithErrors<TodoListItem>> EditItemAsync(TodoListItem replacingItem, IResolvedItem<TodoListItem> existingItem)
+        {
+            if (replacingItem == null)
+            {
+                throw new ArgumentNullException(nameof(replacingItem), "Item to complete must not be null.");
+            }
+
+            if (!existingItem?.WasOperationSuccessful ?? false)
```

Comment:
I meant in https://github.com/petrareb/kentico-onboarding-cs/pull/4#discussion_r270445333 that yours was better 😜 
```suggestion
            if (existingItem == null || !existingItem.WasOperationSuccessful)
```